From 58dfc6a422276429e27a0cf2f468287c4bbadac1 Mon Sep 17 00:00:00 2001
From: Luiz Capitulino <lcapitulino@redhat.com>
Date: Thu, 3 Mar 2011 16:56:37 -0300
Subject: [PATCH 11/12] Revert "ide: Split ide_init1() off ide_init2() (v2)"

RH-Reverts: 7a9b1670759ee9929d7c9595d31be89cec34eb2c
RH-Reverts-patchwork-id: 18376
Bugzilla-related: 677170
Signed-off-by: Luiz Capitulino <lcapitulino@redhat.com>
---
 hw/ide/core.c |   32 +++++++++++++++-----------------
 1 files changed, 15 insertions(+), 17 deletions(-)

diff --git a/hw/ide/core.c b/hw/ide/core.c
index 7522176..54de410 100644
--- a/hw/ide/core.c
+++ b/hw/ide/core.c
@@ -2635,29 +2635,27 @@ void ide_init_drive(IDEState *s, DriveInfo *dinfo, const char *version)
     ide_reset(s);
 }
 
-static void ide_init1(IDEBus *bus, int unit, DriveInfo *dinfo)
-{
-    static int drive_serial = 1;
-    IDEState *s = &bus->ifs[unit];
-
-    s->bus = bus;
-    s->unit = unit;
-    s->drive_serial = drive_serial++;
-    s->io_buffer = qemu_blockalign(s->bs, IDE_DMA_BUF_SECTORS*512 + 4);
-    s->io_buffer_total_len = IDE_DMA_BUF_SECTORS*512 + 4;
-    s->smart_selftest_data = qemu_blockalign(s->bs, 512);
-    s->sector_write_timer = qemu_new_timer(vm_clock,
-                                           ide_sector_write_timer_cb, s);
-    ide_init_drive(s, dinfo, NULL);
-}
-
 void ide_init2(IDEBus *bus, DriveInfo *hd0, DriveInfo *hd1,
                qemu_irq irq)
 {
+    IDEState *s;
+    static int drive_serial = 1;
     int i;
 
     for(i = 0; i < 2; i++) {
-        ide_init1(bus, i, i == 0 ? hd0 : hd1);
+        s = bus->ifs + i;
+        s->bus = bus;
+        s->unit = i;
+        s->drive_serial = drive_serial++;
+        s->io_buffer = qemu_blockalign(s->bs, IDE_DMA_BUF_SECTORS*512 + 4);
+        s->io_buffer_total_len = IDE_DMA_BUF_SECTORS*512 + 4;
+        s->smart_selftest_data = qemu_blockalign(s->bs, 512);
+        s->sector_write_timer = qemu_new_timer(vm_clock,
+                                               ide_sector_write_timer_cb, s);
+        if (i == 0)
+            ide_init_drive(s, hd0, NULL);
+        if (i == 1)
+            ide_init_drive(s, hd1, NULL);
     }
     bus->irq = irq;
 }
-- 
1.7.4.rc1.16.gd2f15e

